src/views/catalog/CreateFromInstanceTypes/components/VMDetailsSection/components/VMDetailsBody/components/DetailsRightGrid.tsx

```diff
@@ -12,23 +12,40 @@ import { humanizeBinaryBytes } from '../../../../../../../../utils/utils/humaniz
 type DetailsRightGridProps = {
   bootSource: V1beta1DataSource;
   namespace: string;
+  storageClassName: string;
+  setStorageClassName: Dispatch<SetStateAction<string>>;
 };
 
-const DetailsRightGrid: React.FC<DetailsRightGridProps> = ({ bootSource, namespace }) => {
+const DetailsRightGrid: React.FC<DetailsRightGridProps> = ({
+  bootSource,
+  namespace,
+  storageClassName,
+  setStorageClassName,
+}) => {
   const [pvcSource, setPVCSource] = useState<V1alpha1PersistentVolumeClaim>();
 
-  const getPVCSource = async () => {
-    if (!bootSource) return;
+  useEffect(() => {
+    if (!bootSource && pvcSource) {
+      setPVCSource(null);
+      return;
+    }
 
-    const pvcData = bootSource?.spec?.source?.pvc;
-    const pvc = await getPVC(pvcData?.name, pvcData?.namespace);
-    setPVCSource(pvc);
-  };
-  getPVCSource().catch(console.log);
+    const pvcMetadata = bootSource?.spec?.source?.pvc;
+    if (
+      pvcSource?.metadata?.name !== pvcMetadata?.name ||
+      pvcSource?.metadata?.namespace !== pvcMetadata?.namespace
+    ) {
+      getPVC(pvcMetadata?.name, pvcMetadata?.namespace)
+        .then((pvc) => {
+          setPVCSource(pvc);
+          setStorageClassName(pvc.spec.storageClassName);
+        })
+        .catch(console.log);
+    }
+  }, [bootSource, pvcSource, setStorageClassName]);
```

Comment:
why this call is in inner component instead of upper component and then u don't need to do prop drilling of 3 floors?

Reply:
not really sure why to be honest, will try improve this, thanks :)

Reply:
made some changes,
now the pvc call is from the outer component but I still had to drill down the pvc for 3 components.
I also think we might want to switch the 5 useState on CreateFromInstanceType.tsx that is related to VM configuration into one useReducer hook. wdyt?

Reply:
yes, please do that, and on the way of doing it, consider better naming of state fields :)

Reply:
I think it will be in a better context to do that in a separated PR